### ISSUE TITLE
[C#] Dividing Null<Int> and assigning it to a double var gives me different result on C#

### ISSUE THREAD

**Issue by vroad** [HaxeFoundation/haxe#5253]
```hx
class Test 
{
	static var a:Null<Int> = 1;
	static var b:Null<Int> = 2;
	
	static function main() 
	{
		var doubleValue:Float = a / b;
		trace(doubleValue);
	}
}
```
This code outputs "0" on C#.
But on other targets I got "0.5". Should C# target convert Null<Int> to double first to get the same result as other targets?

http://try.haxe.org/#dbd71

**Comment by nadako** [comment]
Reproduces on latest git version.

**Comment by nadako** [comment]
This generates integer division (`(global::Main.a).@value / (global::Main.b).@value`), hence the zero instead of 0.5.

This looks bad, can we fix it for 3.3?
